dbms/src/DataStreams/HashJoinProbeBlockInputStream.cpp

```diff
@@ -38,16 +38,29 @@ Block HashJoinProbeBlockInputStream::getTotals()
     if (auto * child = dynamic_cast<IProfilingBlockInputStream *>(&*children.back()))
     {
         totals = child->getTotals();
-        join_probe_actions->executeOnTotals(totals);
+        if (!totals)
+        {
+            if (join && join->hasTotals())
```

Comment:
join must not be nullptr here

Reply:
> join must not be nullptr here

It is checked in constructor of HashJoinProbeBlockStream.

Reply:
yes, I mean that `if (join->hasTotals())` is enough.

Reply:
done.